fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetaResolver.java

```diff
@@ -0,0 +1,183 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.connector.jdbc;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.starrocks.analysis.DateLiteral;
+import com.starrocks.analysis.IntLiteral;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.JDBCResource;
+import com.starrocks.catalog.Table;
+import com.starrocks.catalog.Type;
+import com.starrocks.common.DdlException;
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.connector.ConnectorTableId;
+import com.starrocks.connector.PartitionInfo;
+import com.starrocks.connector.PartitionUtil;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class JDBCMetaResolver implements ConnectorMetadata {
+
+    private static Logger LOG = LogManager.getLogger(JDBCMetaResolver.class);
+
+    private Map<String, String> properties;
+    private String catalogName;
+    private JDBCSchemaResolver schemaResolver;
+
+    private final @NonNull JDBCAsyncCache<JDBCTableName, Integer> tableIdCache = new JDBCAsyncCache<>(true);
+
+    public JDBCMetaResolver(Map<String, String> properties, String catalogName, JDBCSchemaResolver schemaResolver) {
+        this.properties = properties;
+        this.catalogName = catalogName;
+        this.schemaResolver = schemaResolver;
+        checkAndSetSupportPartitionInformation();
+    }
+
+    public Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(properties.get(JDBCResource.URI),
+                properties.get(JDBCResource.USER), properties.get(JDBCResource.PASSWORD));
+    }
+
+    public void checkAndSetSupportPartitionInformation() {
+        try (Connection connection = getConnection()) {
+            schemaResolver.checkAndSetSupportPartitionInformation(connection);
+        } catch (SQLException e) {
+            throw new StarRocksConnectorException(e.getMessage());
+        }
+    }
+
+    List<String> refreshCacheForListDbNames() {
+        try (Connection connection = getConnection()) {
+            return Lists.newArrayList(schemaResolver.listSchemas(connection));
+        } catch (SQLException e) {
+            throw new StarRocksConnectorException(e.getMessage());
```

Comment:
it's best to print error stack to locate issue.

Reply:
replace with 'throw new StarRocksConnectorException("refresh db names cache for JDBC catalog fail!", e);'